src/ClassMapGenerator.php

```diff
@@ -142,18 +142,20 @@ public function scanPaths($path, ?string $excluded = null, string $autoloadType
                 continue;
             }
 
-            if (!self::isAbsolutePath($filePath)) {
+            if (!self::isAbsolutePath($filePath) && !self::isStreamWrapper($filePath)) {
                 $filePath = $cwd . '/' . $filePath;
                 $filePath = self::normalizePath($filePath);
-            } else {
+            } elseif(!self::isStreamWrapper($filePath)) {
                 $filePath = Preg::replace('{[\\\\/]{2,}}', '/', $filePath);
```

Comment:
```suggestion
            } else {
                $filePath = Preg::replace('{(?<!:)[\\\\/]{2,}}', '/', $filePath);
```

This should prevent foo:// from being replaced